src/pmhn/_trees/_io_modified.py

```diff
@@ -0,0 +1,113 @@
+"""Utilities for parsing data frames into AnyTree trees."""
+import dataclasses
+from typing import Any
+
+import anytree
+import pandas as pd
+
+
+@dataclasses.dataclass
+class TreeNaming:
+    """Naming conventions used to parse a tree.
+
+    Attrs:
+        node: column name with node id/name
+        parent: column name with the parent's id
+        data: a dictionary mapping column names to field names in nodes
+
+    Example:
+    TreeNaming(
+        node="Node_ID",
+        parent="Parent_ID",
+        data={
+            "Mutation_ID": "mutation",
+            "SomeValue": "value",
+        }
+    means that a data frame with columns
+    "Node_ID", "Parent_ID", "Mutation_ID", "SomeValue"
+    is expected.
+
+    Created nodes will have additional fields
+    "mutation" and "value".
+    """
+
+    node: str = "Node_ID"
+    parent: str = "Parent_ID"
+    data: dict[str, str] = dataclasses.field(
+        default_factory=lambda: {"Mutation_ID": "mutation"}
+    )
+
+
+@dataclasses.dataclass
+class ForestNaming:
+    """Naming conventions used to parse a forest (a set of trees).
+
+    Attrs:
+        tree_name: column name storing the tree id/name
+        naming: TreeNaming object used to parse each tree
+    """
+
+    tree_name: str = "Tree_ID"
+    naming: TreeNaming = dataclasses.field(default_factory=TreeNaming)
+
+
+def parse_tree(df: pd.DataFrame, naming: TreeNaming) -> anytree.Node:
+    """Parses a data frame into a tree
+
+    Args:
+        df: data frame with columns specified in `naming`.
+        naming: specifies the columns that should be present in `df`
+
+    Returns:
+        the root node of the tree
+    """
+    root = None
+    nodes = {}  # Maps a NodeID value to Node
+
+    for _, row in df.iterrows():
+        node_id = row[naming.node]
+        parent_id = row[naming.parent]
+        values = {val: row[key] for key, val in naming.data.items()}
+
+        if node_id in nodes:
+            raise ValueError(f"Node {node_id} already exists.")
+
+        # We found the root
+        if node_id == parent_id:
+            if root is not None:
+                raise ValueError(
+                    f"Root is {root}, but {node_id} == {parent_id} "
+                    "also looks like a root."
+                )
+            root = anytree.Node(values["mutation"], parent=None, **values)
```

Comment:
This file differs form `_io.py` by two lines of code. [Duplicate code](https://en.wikipedia.org/wiki/Duplicate_code) is generally considered an antipattern.

I propose to remove this file entirely and change these two lines `_io` directly :slightly_smiling_face: 
(They will work, nice job!)